src/frontend/lexer.rs

```diff
@@ -0,0 +1,259 @@
+use itertools::Itertools;
+use std::iter::Peekable;
+
+use super::token::Token;
+
+// -------------------------------------------------------------------------- //
+// Error type                                                                 //
+// -------------------------------------------------------------------------- //
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Error {
+    UnexpectedSymbol(char),
+    UnclosedString,
+}
+
+// -------------------------------------------------------------------------- //
+// Type aliases                                                               //
+// -------------------------------------------------------------------------- //
+
+type TokResult = Result<Token, Error>;
+
+// -------------------------------------------------------------------------- //
+// Lexer struct                                                               //
```

Comment:
A lot of these comments feel redundant on first glance. A good practice imo is to have self-documenting code (which you've done an excellent job with) and reserve comments for "black-magic" code that is impossible to describe with a good function/variable name OR public apis that come with examples/doctests .

What do you think?

Reply:
My intent with these comments was to provide a higher level overview of the source file. I haven't worked on anything beyond personal projects and those never got large. 

I guess I'm kind of treating them as C# [regions](https://www.dotnetperls.com/region). But yeah, they aren't useful :)